avm/res/batch/batch-account/main.bicep

```diff
@@ -0,0 +1,403 @@
+metadata name = 'Batch Accounts'
+metadata description = 'This module deploys a Batch Account.'
+metadata owner = 'Azure/module-maintainers'
+
+@description('Required. Name of the Azure Batch.')
+param name string
+
+@description('Optional. Location for all Resources.')
+param location string = resourceGroup().location
+
+@description('Required. The resource ID of the storage account to be used for auto-storage account.')
+param storageAccountId string
+
+@allowed([
+  'BatchAccountManagedIdentity'
+  'StorageKeys'
+])
+@description('Optional. The authentication mode which the Batch service will use to manage the auto-storage account.')
+param storageAuthenticationMode string = 'StorageKeys'
+
+@description('Optional. The resource ID of a user assigned identity assigned to pools which have compute nodes that need access to auto-storage.')
+param storageAccessIdentityResourceId string?
+
+@allowed([
+  'BatchService'
+  'UserSubscription'
+])
+@description('Optional. The allocation mode for creating pools in the Batch account. Determines which quota will be used.')
+param poolAllocationMode string = 'BatchService'
+
+@description('Conditional. The key vault to associate with the Batch account. Required if the \'poolAllocationMode\' is set to \'UserSubscription\' and requires the service principal \'Microsoft Azure Batch\' to be granted contributor permissions on this key vault.')
+param keyVaultReferenceResourceId string?
+
+@description('Optional. Configuration details for private endpoints. For security reasons, it is recommended to use private endpoints whenever possible.')
+param privateEndpoints privateEndpointType
+
+@description('Optional. Whether or not public network access is allowed for this resource. For security reasons it should be disabled. If not specified, it will be disabled by default if private endpoints are set and networkProfileAllowedIpRanges are not set.')
+@allowed([
+  ''
+  'Enabled'
+  'Disabled'
+])
+param publicNetworkAccess string = ''
+
+@allowed([
+  'Allow'
+  'Deny'
+])
+@description('Optional. The network profile default action for endpoint access. It is only applicable when publicNetworkAccess is not explicitly disabled.')
+param networkProfileDefaultAction string = 'Deny'
```

Comment:
What about a `networkProfile` object hosting this and the next parameter?

Reply:
In that case we'd need another User Defined Type. Not saying we shouldn't, but this would need to be updated in all modules with PE (+ specs) and I suggest to do this  in a separate issue + PR.

Reply:
Do we have a spec saying objects cannot be used unless typed with udt? I agree that would be the end goal but for the moment we made exceptions, e.g. for child resources, and not only. Other modules have networkAcls in an object, without udt (e.g. cognitive services). NetworkAcls and networkProfile are basically the same thing. So yes, let's agree on something.

Reply:
For the time being, I would only like to enforce UDT for what we prepared with extension resources. If somebody wants to go the extra mile and add any UDT on top of that, be my guest (e.g., @ChrisSidebotham). But blocking PRs for that seems wrong to me as we have more than enough work porting the CARML modules as is, yet alone optimizing them for UDTs.

Reply:
I'm not asking for UDT, I'm suggesting to use an object for networkProfile, to align to what done for networkAcls for other modules. Adding udt next will be easier if we already have an object.

Reply:
I see. Sorry I'm only working of the comment. I'll compare the two and check it out

Reply:
Turns out it's actually a bit more different that I thought. May I suggest that we push the change to the module's logic back?

Reply:
OK, but I'll raise one PR to fix this and bump to 0.2.0 before getting this ORPHANED if that is OK

Reply:
Done in PR #592